microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -81,28 +69,75 @@ def _execute_expect_zero(self, node: Node, cmd: str) -> str:
         result.assert_exit_code()
         return result.stdout
 
-    def _hugepages_init(self, node: Node) -> None:
-        mount = node.tools[Mount]
-        mount.mount(disk_name="nodev", point="/mnt/huge", type="hugetlbfs")
-        mount.mount(
-            disk_name="nodev",
-            point="/mnt/huge-1G",
-            type="hugetlbfs",
-            options="pagesize=1G",
-        )
+    @TestCaseMetadata(
+        description="""
+           Tests a more realistic sender/forwarder/receiver setup.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            min_core_count=8,
+            min_nic_count=2,
+            network_interface=Sriov,
+            min_count=3,
+        ),
+    )
+    def check_3node_forwarding(self, environment: Environment, log: Logger) -> None:
+        # TODO: stub test, just submitting refactoring PR
+        pass
+
+
+def hugepages_init(node: Node) -> None:
+    mount = node.tools[Mount]
+    mount.mount(disk_name="nodev", point="/mnt/huge", type="hugetlbfs")
+    mount.mount(
+        disk_name="nodev",
+        point="/mnt/huge-1G",
+        type="hugetlbfs",
+        options="pagesize=1G",
+    )
 
-    def _hugepages_enable(self, node: Node, log: Logger) -> None:
-        echo = node.tools[Echo]
-        echo.write_to_file(
-            "1024",
-            "/sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages",
-            sudo=True,
-        )
-        echo.write_to_file(
-            "1",
-            "/sys/devices/system/node/node0/hugepages/hugepages-1048576kB/nr_hugepages",
-            sudo=True,
-        )
+
+def hugepages_enable(node: Node) -> None:
+    echo = node.tools[Echo]
+    echo.write_to_file(
+        "1024",
+        "/sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages",
+        sudo=True,
+    )
+    echo.write_to_file(
+        "1",
+        "/sys/devices/system/node/node0/hugepages/hugepages-1048576kB/nr_hugepages",
+        sudo=True,
+    )
+
+
+class DpdkTestKit:
+    def __init__(self, node: Node, log: Logger) -> None:
+        network_interface_feature = node.features[NetworkInterface]
```

Comment:
Don't put the long time logic in construction method like `__init__`.

Reply:
Not clear on what the change needs to be, you want things that will run for a while in a separate function?
kit = TestKit()
change to like 
kit = TestKit()
kit.init_long_running_stuff()

Reply:
Yes, put it in a separated method. Python may not be the same, but in most OO languages, it should avoid to have logic in construction.

Reply:
I think it's more pythonic to put all the init steps into __init__. Any error during this step isn't recoverable so the only effect of moving it out out __init__ is that we now have to write two lines instead of one everywhere.

Reply:
In Python code, if it has one method, it can be a function, instead of a class. Forgot to give this suggestion in first round...